### ISSUE TITLE
Question: network resolution

### ISSUE THREAD

**Issue by HUNGLiquid** [AlexeyAB/darknet#2101]
Hi @AlexeyAB, I have a question about network resolution.
When I trained, my network resolution is 416x416. I changed it to 320x320, 416x416 and 608x608 in testing phase. I realize results is not reasonable. 
320x320 false positive and false negative are high.
416x416 false positive and false negative are lower than.
608x608 has a problem. Sometimes it is better, FP and FN are lower than 416x416, but sometimes are higher. Why does it happen?

**Comment by AlexeyAB** [comment]
@HUNGLiquid 

* Did you train by using 416x416 random=1 in cfg-file?

* Did you measure FP and FN by yourself or by using ./darknet detector map ... command?

* Can you show screenshot of output of command `./darknet detector map ...` for 320x320, 416x416, 608x608?

* In general the higher resolution - the higher accuracy mAP (lower FP, FN and higher TP). But if you trained on 320x320 with random=0, and then test it with 608x608 then accuracy can be lower, beceause there can be broken this rule:  https://github.com/AlexeyAB/darknet#how-to-improve-object-detection


> General rule - your training dataset should include such a set of relative sizes of objects that you want to detect:
> 
> * `train_network_width * train_obj_width / train_image_width ~= detection_network_width * detection_obj_width / detection_image_width`
> * `train_network_height * train_obj_height / train_image_height ~= detection_network_height * detection_obj_height / detection_image_height`
> 
> I.e. for each object from Test dataset there must be at least 1 object in the Training dataset with about the same relative size:
> 
> `object width in percent from Training dataset ~= object width in percent from Test dataset`
> 
> That is, if only objects that occupied 80-90% of the image were present in the training set, then the trained network will not be able to detect objects that occupy 1-10% of the image.

**Comment by HUNGLiquid** [comment]
@AlexeyAB 
1. Yes. I trained by using 416x416 random=1 in cfg-file.
2. I measured FP and FN by myself for each class.
3. But I can use the command:
608x608
![image](https://user-images.githubusercontent.com/32360741/50424250-e19eb480-0893-11e9-9113-eee0022f3b95.png)
416x416
![image](https://user-images.githubusercontent.com/32360741/50424263-16127080-0894-11e9-8bd7-cc810d7aaba9.png)
But when I detect with threshold=0.5 and I saw results 608 is lower. FN of class "shoe"=82 when 416 just has 5. 
![image](https://user-images.githubusercontent.com/32360741/50424352-fd568a80-0894-11e9-9606-bc8a93de75d0.png)
4. I didn't follow the rule because I have 6 classes of different sizes.

**Comment by AlexeyAB** [comment]
> When I detect objects **with other backgrounds** (not training or training with old weight and more 300 iterations), results are not good. How can I improve it?

You should add images with these backgrounds to the training dataset: https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

You should have about 12 000 different training images for 6 objects.

> * desirable that your training dataset include images with objects at diffrent: scales, rotations, lightings, from different sides, on different backgrounds - you should preferably have 2000 different images for each class or more, and you should train 2000*classes iterations or more
> 
> * desirable that your training dataset include images with non-labeled objects that you do not want to detect - negative samples without bounded box (empty .txt files) - use as many images of negative samples as there are images with objects
